### PR TITLE
Add support for more JSON Schema specs

### PR DESCRIPTION
**Pull request by dfernandez-at-wiris**
- add checks for drafts 4, 6 and 7, defaulting to 7 if unspecified
- simplify exceptions, reduce them in number
- add a minor functionality to list the available configurations
- minor docstring and README.md corrections/expansions
- add the appropriate unit tests & fixtures for the added functionality

### CODE REVIEW

File: santoku/utils/README.md [hunk 1]
```diff
@@ -7,44 +7,38 @@ The `configuration` module provides classes to store and manage configuration fi
...
-    ["object_setting", "nested_boolean_setting"]
-)
+# nested settings can be accessed by calling get_setting() on the list of keys, in order
+nested_boolean_setting = configuration_manager.get_setting("object_setting", "nested_setting")
 ```
 
 
 ### Schema validation
-For schema validation of configuration files, we use our own definition of what a configuration looks like and also our own definition of what a schema should look like. In short, our schema is a narrow subset of [JSON Schema](https://json-schema.org/). We plan on fully adopting JSON Schema for our validation needs, but due to the lack of native Python tools and the effort of implementing our own parser we erred on the side of something simpler that allows for faster iteration.
+For schema validation of configuration files, we use [JSON Schema](https://json-schema.org/) and the [jsonschema](https://pypi.org/project/jsonschema/) package. For a
```
Comment (hqiu-at-wiris) [inline santoku/utils/README.md]: Is this "For a" a typo?

File: santoku/utils/configuration.py [hunk 2]
```diff
@@ -137,13 +122,15 @@ def __init__(
         initial_configuration: str, optional
             Name of the configuration in the `configurations` that will be set as the initial one.
 
-        InvalidConfiguration
```
Comment (hqiu-at-wiris) [inline santoku/utils/configuration.py:140]: Missing `Raises` section title.

File: santoku/utils/configuration.py [hunk 3]
```diff
@@ -2,7 +2,6 @@
 
 import jsonschema
 
-from jsonschema import Draft7Validator
 from jsonschema.exceptions import ValidationError, SchemaError
 from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
```
Comment (hqiu-at-wiris) [inline santoku/utils/configuration.py]: `Iterable` and `Tuple` not needed.

**Review by hqiu-at-wiris — APPROVED** [review]

File: santoku/utils/test/test_configuration.py [hunk 4]
```diff
@@ -5,13 +5,13 @@
 from collections import namedtuple
```
Comment (hqiu-at-wiris) [inline santoku/utils/test/test_configuration.py]: Not used anywhere.

File: santoku/utils/README.md (hunk 1 above)
Comment (dfernandez-at-wiris) [reply santoku/utils/README.md]: Indeed

**Pull request merged**